sfepy/homogenization/coefs_base.py

```diff
@@ -737,20 +739,26 @@ def compute_correctors(self, evp, sign, state0, ts, dump_name, save_name,
 ##             print aaa
 ##             print bbb
 
+            state[vu].append(vec_u)
+            state[vp].append(vec_p)
+            state['d' + vp].append(vec_dp)
+
             self.save(dump_name, save_name, vec_u, vec_p, vec_dp, ts, problem)
 
+        return state
+
     def save(self, dump_name, save_name, vec_u, vec_p, vec_dp, ts, problem):
         """
         1. saves raw correctors into hdf5 files (filename)
         2. saves correctors transformed to output for visualization
         """
         vu, vp = self.dump_variables
-        out = {vu : Struct(name='dump', mode='nodes', data=vec_u,
-                           dofs=None, var_name=vu),
-               vp : Struct(name='dump', mode='nodes', data=vec_p,
-                           dofs=None, var_name=vp),
-               'd'+vp : Struct(name='dump', mode='nodes', data=vec_dp,
-                               dofs=None, var_name=vp)}
+        out = {vu : Struct(name='dump', mode='nodes', dofs=None,
```

Comment:
Maybe also remove the dofs arg (not used for anything), and fix `mode='nodes'` to `mode='vertex'`.

Reply:
OK